src/figuresClasses.ts

```diff
@@ -1,11 +1,70 @@
-export interface Figure {}
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
 
-export class Triangle implements Figure {}
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+    public shape: Shape = 'triangle',
+  ) {
+    const sideOfTriangle = [a, b, c].sort((a1, b1) => b1 - a1);
+    const condition =
+      sideOfTriangle[0] >= sideOfTriangle[1] + sideOfTriangle[2];
+
+    if (a <= 0 || b <= 0 || c <= 0 || condition) {
+      throw new Error('your error message');
+    }
+  }
+
+  getArea(): number {
+    const s = (1 / 2) * (this.a + this.b + this.c);
+    const area = Math.sqrt(s * (s - this.a) * (s - this.b) * (s - this.c));
+
+    return Math.round(area * 100) / 100;
+  }
+}
+
+export class Circle implements Figure {
+  constructor(
+    public color: Color,
+    public radius: number,
+    public shape: Shape = 'circle',
+  ) {
+    if (radius <= 0) {
+      throw new Error('your error message');
+    }
+  }
 
-export class Circle implements Figure {}
+  getArea(): number {
+    return Math.floor(this.radius ** 2 * Math.PI * 100) / 100;
+  }
+}
 
-export class Rectangle implements Figure {}
+export class Rectangle implements Figure {
+  constructor(
+    public color: Color,
+    public width: number,
+    public height: number,
+    public shape: Shape = 'rectangle',
+  ) {
+    if (width <= 0 || height <= 0) {
+      throw new Error('your error message');
```

Comment:
The error message should be detailed to explain why the width or height values are invalid.